server/api/initial_data.py

```diff
@@ -894,6 +903,34 @@ def _create_project_summaries(db, db_session):
     db._upsert(db_session, project_summaries, ignore=True)
 
 
+def _align_schedule_labels(db, db_session):
+    db_schedules: list[mlrun.common.schemas.ScheduleRecord] = db.list_schedules(
+        session=db_session
+    )
+    schedules_update = []
+    for db_schedule in db_schedules:
+        # convert list[LabelRecord] to dict
+        db_schedule_labels = {label.name: label.value for label in db_schedule.labels}
+        # merging labels
+        merged_labels = server.api.utils.scheduler.Scheduler()._merge_schedule_and_schedule_object_labels(
+            labels=db_schedule_labels,
+            scheduled_object=db_schedule.scheduled_object,
+        )
+
+        # get a Schedule object (not a ScheduleRecord) and update it
+        schedule = db._get_schedule_record(
+            db_session, db_schedule.project, db_schedule.name
+        )
+        db._update_schedule_body(
+            schedule=schedule,
+            scheduled_object=db_schedule.scheduled_object,
+            labels=merged_labels,
+        )
+        schedules_update.append(schedule)
+
+    db._upsert(db_session, schedules_update)
```

Comment:
1. This method is very expensive. It is going to preform `O(n)` queries for each schedule in the DB. I suggest adding a `as_records: bool = False,` param to `list_schedules` (similar to `list_artifacts`). You can still extract the schedule struct with `_transform_schedule_record_to_scheme` but then you don't need to re-query for each record after getting them all.
2. This method is using loads of the `db`'s private methods, I don't think we should make them public, but maybe this method should be part of the `sqldb` interface?